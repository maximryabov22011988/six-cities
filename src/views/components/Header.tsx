import * as React from 'react';


const propTypes = {
  logo: PropTypes.element,
  userInfo: PropTypes.element,
};

function Header({ logo, userInfo }) {
  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">{logo}</div>
          <nav className="header__nav">
            <ul className="header__nav-list">{userInfo}</ul>
          </nav>
        </div>
      </div>
    </header>
  );
}

Header.propTypes = propTypes;

export default Header;
