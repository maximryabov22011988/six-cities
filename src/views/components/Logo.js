import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classnames from 'classnames';

const propTypes = {
  position: PropTypes.string,
  src: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
  label: PropTypes.string,
  isActive: PropTypes.bool,
};

const defaultProps = {
  src: 'img/logo.svg',
  width: '81',
  height: '41',
  label: '6 cities logo',
};

class Logo extends React.PureComponent {
  render() {
    const { position, src, width, height, label, isActive } = this.props;

    const linkClasses = classnames(
      position && `${position}__logo-link`,
      isActive && `${position}__logo-link--active`,
    );
    const logoClasses = classnames(position && `${position}__logo`);

    return isActive ? (
      <div className={linkClasses}>
        <img
          className={logoClasses}
          src={src}
          width={width}
          height={height}
          alt={label}
        />
      </div>
    ) : (
      <Link to="/" className={linkClasses}>
        <img
          className={logoClasses}
          src={src}
          width={width}
          height={height}
          alt={label}
        />
      </Link>
    );
  }
}

Logo.propTypes = propTypes;
Logo.defaultProps = defaultProps;

export default Logo;
