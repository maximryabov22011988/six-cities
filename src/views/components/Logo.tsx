import * as React from 'react';
import { Link } from 'react-router-dom';
import cn from 'classnames';

interface Props {
  height?: string;
  isActive?: boolean;
  label?: string;
  position: string;
  src?: string;
  width?: string;
}

class Logo extends React.PureComponent<Props> {
  render() {
    const {
      position,
      src = 'img/logo.svg',
      width = '81',
      height = '41',
      label = '6 cities logo',
      isActive,
    } = this.props;

    const linkClasses = cn(position && `${position}__logo-link`, isActive && `${position}__logo-link--active`);
    const logoClasses = cn(position && `${position}__logo`);

    return isActive ? (
      <div className={linkClasses}>
        <img alt={label} className={logoClasses} height={height} src={src} width={width} />
      </div>
    ) : (
      <Link className={linkClasses} to="/">
        <img alt={label} className={logoClasses} height={height} src={src} width={width} />
      </Link>
    );
  }
}

export default Logo;
