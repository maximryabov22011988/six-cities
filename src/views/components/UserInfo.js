import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { BASE_URL } from '../../api';

const propTypes = {
  isAuth: PropTypes.bool.isRequired,
  avatarUrl: PropTypes.oneOfType([PropTypes.string, PropTypes.any]),
  email: PropTypes.oneOfType([PropTypes.string, PropTypes.any]),
};

class UserInfo extends React.PureComponent {
  render() {
    const { isAuth, avatarUrl, email } = this.props;

    return (
      <li className="header__nav-item user">
        {!isAuth ? (
          <Link to="/login">Sign In</Link>
        ) : (
          <Link
            to="/favorites"
            className="header__nav-link header__nav-link--profile"
          >
            <div className="header__avatar-wrapper user__avatar-wrapper">
              <img
                className="user__avatar"
                alt="avatar"
                src={`${BASE_URL}${avatarUrl}`}
              />
            </div>
            <span className="header__user-name user__name">{email}</span>
          </Link>
        )}
      </li>
    );
  }
}

UserInfo.propTypes = propTypes;

export default UserInfo;
