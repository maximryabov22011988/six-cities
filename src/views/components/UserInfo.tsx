import * as React from 'react';

import { Link } from 'react-router-dom';

import Avatar from './Avatar';

import { BASE_URL } from '../../api';

const propTypes = {
  avatarUrl: PropTypes.oneOfType([PropTypes.string, PropTypes.any]),
  email: PropTypes.oneOfType([PropTypes.string, PropTypes.any]),
  isAuth: PropTypes.bool.isRequired,
};

class UserInfo extends React.PureComponent {
  render() {
    const { isAuth, avatarUrl, email } = this.props;

    return (
      <li className="header__nav-item user">
        {!isAuth ? (
          <Link to="/login">Sign In</Link>
        ) : (
          <Link className="header__nav-link header__nav-link--profile" to="/favorites">
            <Avatar alt="User avatar" name={email} parentClassName="header" src={`${BASE_URL}${avatarUrl}`} />
          </Link>
        )}
      </li>
    );
  }
}

UserInfo.propTypes = propTypes;

export default UserInfo;
