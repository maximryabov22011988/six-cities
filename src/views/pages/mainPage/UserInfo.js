import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const propTypes = {
  isAuth: PropTypes.bool.isRequired,
  email: PropTypes.string.isRequired
};

class UserInfo extends React.PureComponent {
  render() {
    const { isAuth, email } = this.props;
    return (
      <li className="header__nav-item user">
        {!isAuth ? (
          <Link to="/login">Sign In</Link>
        ) : (
          <a className="header__nav-link header__nav-link--profile" href="#">
            <div className="header__avatar-wrapper user__avatar-wrapper" />
            <span className="header__user-name user__name">{email}</span>
          </a>
        )}
      </li>
    );
  }
}

UserInfo.propTypes = propTypes;

export default UserInfo;
