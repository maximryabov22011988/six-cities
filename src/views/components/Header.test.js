import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';
import Logo from './Logo';
import UserInfo from '../pages/mainPage/UserInfo';

const userMock = {
  isAuth: true,
  avatarUrl: '/static/avatar/1.jpg',
  email: 'Oliver.conner@gmail.com'
};

it('Header correctly render', () => {
  const header = renderer
    .create(
      <Header
        logo={<Logo />}
        userInfo={
          <UserInfo
            isAuth={userMock.isAuth}
            avatarUrl={userMock.avatarUrl}
            email={userMock.email}
          />
        }
      />
    )
    .toJSON();

  expect(header).toMatchSnapshot();
});
