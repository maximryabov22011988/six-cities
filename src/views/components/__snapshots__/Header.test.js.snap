// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header correctly render 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <div
              className="header__nav-link header__nav-link--profile"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt="avatar"
                  className="user__avatar"
                  src="https://es31-server.appspot.com/six-cities/static/avatar/1.jpg"
                />
              </div>
              <span
                className="header__user-name user__name"
              >
                Oliver.conner@gmail.com
              </span>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
